hermes/validator/challenge_manager.py

```
@@ -594,27 +623,111 @@ async def query_miner(
+                        self._last_set_weight_time = time.time()
                         continue
-                    scores = [s[0] for s in scores]
-                    self._set_weights(uids, scores)
+                    logger.info("[ChallengeManager] No historical scores available. Submitting uniform fallback weights.")
+
+                self._set_weights(uids, scores)
+                self._last_set_weight_time = time.time()
+                if epoch_info:
+                    self._last_epoch_submitted = epoch_info.epoch_index
+                logger.info(f"[ChallengeManager] Submitted weights due to {reason}.")
+            except Exception as e:
```

Ruff (0.14.13)

Do not catch blind exception: `Exception`

(BLE001)